File: liqp-core/src/main/java/liqp/nodes/AtomNode.java
package liqp.nodes;

import java.util.Collections;
import java.util.List;
import liqp.context.LContext;
import liqp.node.LNode;

public class AtomNode implements LNode {

    private Object value;

    public AtomNode(Object value) {
        this.value = value;
    }


    @Override
    public Object render(LContext context) {

        return value;
    }

    @SuppressWarnings("unchecked")
    public <X> X get() {
      return (X) value;
    }

  @Override
  public List<LNode> children() {
    return Collections.emptyList();
  }
}

File: liqp-core/src/main/kotlin/liqp/nodes/AtomNode.kt
@file:Suppress("UNCHECKED_CAST")

package liqp.nodes

import liqp.context.LContext
import liqp.node.LNode

class AtomNode(private val value: Any?) : LNode() {

  override val children: List<LNode> = emptyList()

  override fun render(context: LContext): Any? {
    return value
  }

  fun <X> get(): X {
    return value as X
  }
}
